src/mapping.ts

```diff
@@ -280,7 +280,8 @@ export function handleShelfRepay(call: BorrowCall): void {
   loan.repaysAggregatedAmount = loan.repaysAggregatedAmount.plus(amount)
   loan.repaysCount = loan.repaysCount + 1
   // decrease debt here. Reason: debt won't be updated on every block, but we want relatively up-to-date information in
-  // the UI
+  // the UI. Note that debt should not get negative, so we decrease debt (and totalDebt below) maximally by loan.debt
+  let debtDecrease = amount.gt(loan.debt) ? loan.debt : amount
   loan.debt = loan.debt.minus(amount)
```

Comment:
let's dicrease the loan.debt & totalDebt by the debtDecrease & not amount anymore :)

Reply:
Good find ;) 
Fixed!